core/services/job/orm.go

```diff
@@ -206,7 +206,15 @@ func (o *orm) CreateJob(ctx context.Context, jobSpec *Job, p pipeline.Pipeline)
 		}
 		jobSpec.DirectRequestSpecID = &jobSpec.DirectRequestSpec.ID
 	case FluxMonitor:
-		err := tx.Create(&jobSpec.FluxMonitorSpec).Error
+		existing := FluxMonitorSpec{}
+		err := o.db.First(&existing, "contract_address = ?", jobSpec.FluxMonitorSpec.ContractAddress).Error
```

Comment:
oh could we just put a unique constraint on the contract_address field in the `flux_monitor_specs` ?

Reply:
hmm not sure, would not be backwards compatible with existing data, and migrating it might be tricky

Reply:
With multi chain in place it seems possible that you could have the same contract address on different chains?

Reply:
Yeah itd have to be UNIQUE(contract_address, evm_chain_id) @jkongie good point. @PiotrTrzpil I think we could just fail the upgrade/migration if someone has duplicates and throw an error message telling them to delete all but one should they erroneously have dupes